src/Transpiler.ts

```diff
@@ -1030,13 +1030,21 @@ export abstract class LuaTranspiler {
         return parts.join("..");
     }
 
-    public transpileConditionalExpression(node: ts.ConditionalExpression, brackets?: boolean): string {
+    public transpileProtectedConditionalExpression(node: ts.ConditionalExpression): string {
         const condition = this.transpileExpression(node.condition);
         const val1 = this.transpileExpression(node.whenTrue);
         const val2 = this.transpileExpression(node.whenFalse);
+        return `((${condition}) and function() return ${val1}; end or function() return ${val2}; end)()`;
+    }
 
-        return this.transpileLuaLibFunction(LuaLibFeature.Ternary, condition,
-                                            `function() return ${val1} end`, `function() return ${val2} end`);
+    public transpileConditionalExpression(node: ts.ConditionalExpression): string {
+        if (tsHelper.isFalsible(this.checker.getTypeAtLocation(node.whenTrue), this.options.strictNullChecks)) {
```

Comment:
One more thing (sorry): "strictNullChecks" is implicitly enabled if "strict" is set in options, so you'll want to check for that too.

Reply:
Alright.